### PR TITLE
recharts: add missing label definitions to BarProps XAxisProps and YAxisProps

### PR DESCRIPTION
**Pull request by marcelowa**

### CODE REVIEW

**Comment by mthmulders** [comment]
Thanks for your contribution!

It doesn't look too complicated. Can you add a reference to Recharts documentation or source code that illustrates the `label` is there? Could you also add/extend a test case illustrating its usage?

**Comment by marcelowa** [comment]
Should I add links to the documentation above the added lines?
Notice the documentation isn't very clear regarding the shape of these labels, I had to try and fail until I figured it, but at least now it's supported in the definition

Regarding the tests, should I just add it to the `recharts-tests.tsx` render function ? I don't see any assertions whatsoever there

**Comment by mthmulders** [comment]
It would've been fine with me if you'd just update this issue with the documentation links :).

Indeed, you can update `recharts-tests.tsx`. As far as I can see, the usual DefinitelyTyped approach is to have a test that basically tries to transpile the JSX-file against the current type definitions. If that succeeds, the test is green.

File: types/recharts/index.d.ts [hunk 1]
```diff
@@ -868,6 +878,8 @@ export interface XAxisProps extends EventAttributes {
     tickSize?: number;
     interval?: AxisInterval;
     reversed?: boolean;
+    // see label section at http://recharts.org/#/en-US/api/XAxis
```
Comment (minestarks) [inline types/recharts/index.d.ts]: The doc links look a bit out of place when none of the other properties have it, I'd suggest removing them and following @mthmulders 's advice of adding test cases. Thanks.

**Review by minestarks — CHANGES_REQUESTED** [review]

**Comment by ZheyangSong** [comment]
As one of the many (https://github.com/recharts/recharts/blob/c8bda715232b3bb0215d5bd4d85168e4b8cc5b55/src/polar/Pie.js#L55-L60), the label's shape is described at the level where the chart component is implemented.

According to xAxis's [spec](https://github.com/recharts/recharts/blob/709b1da5d870de1dc52f6225acf5f6a207593be1/test/specs/cartesian/XAxisSpec.js) and its [source code](https://github.com/recharts/recharts/blob/master/src/cartesian/XAxis.js), I don't find `label` is defined or used. Probably I missed something?

**Pull request merged**